### DISCUSSION TITLE
Data Virtualization

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#11534] [Feature Requests]
@benhaynes hi Ben - are there any plans in the future to implement data virtualization across multiple sources? 

A good example I came recently across is Denodo… might be worth to have a look at it if this is something you want to strategically go for…

KR
Roman

**Comment by benhaynes** [comment]
Yes, this is a bigger lift, and would probably be part of a major update (eg: Directus 10), but we're interested in looking at how we would support multiple databases/data-sources at once. The big things to look into are naming conflicts across sources (would need to scope collections to each store) and how that would work in our current API spec... as well as if/how we would handle relationships across different datastores. UUIDs help, but we couldn't leverage the power of native SQL joins... especially if any sources were NoSQL.

It's a big shift, so we're going to get what we have (one SQL database) really dialed in first. 👍

**Reply by benhaynes** [reply]
Also, we are working on a native no-code way to include data from other external sources in the Directus API. Big stuff on the horizon.

**Reply by ernst86913** [reply]
@benhaynes cool - as mentioned - have a look at the community edition of Denodo… you might find some ideas there.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
